src/data/providers/balancer-api/client/index.ts

```diff
@@ -0,0 +1,31 @@
+import { ChainId } from '../../../../utils';
+
+export class BalancerApiClient {
+    apiUrl: string;
+    chainId: ChainId;
+    constructor(apiUrl: string, chainId: number) {
+        this.apiUrl = apiUrl;
+        this.chainId = chainId;
+    }
+
+    async fetch(
```

Comment:
Is this meant to be flexible to allow different types of querys (apart from GetPool)? `variables: { id: string }` would probably limit that?

Reply:
How about changing so `BalancerApiClient.fetch` accepts the JSON query as the only param and `Pools.fetchPoolState` is responsible for putting that together?

Reply:
This is a good idea, it will have only the requestQuery as a parameter on the fetch